extensions/interactions/CodeRepl/directives/CodeRepl.js

```diff
@@ -21,16 +21,24 @@
  */
 oppia.directive('oppiaInteractiveCodeRepl', [
   'HtmlEscaperService', 'codeReplRulesService', 'UrlInterpolationService',
-  function(HtmlEscaperService, codeReplRulesService, UrlInterpolationService) {
+  'EVENT_NEW_CARD_AVAILABLE',
+  function(HtmlEscaperService, codeReplRulesService, UrlInterpolationService,
+      EVENT_NEW_CARD_AVAILABLE ) {
     return {
       restrict: 'E',
       scope: {
-        onSubmit: '&'
+        onSubmit: '&',
+        getLastAnswer: '&lastAnswer'
       },
       templateUrl: UrlInterpolationService.getExtensionResourceUrl(
         '/interactions/CodeRepl/directives/' +
         'code_repl_interaction_directive.html'),
       controller: ['$scope', '$attrs', function($scope, $attrs) {
+        $scope.interactionIsActive = ($scope.getLastAnswer() === null) ||
```

Comment:
I think this should be more deterministic. Why is there an "undefined" case? Can we change things so that getLastAnswer() is always defined?

Maybe one way is to bundle all the init() stuff into an init() method and call that after a short timeout.

Reply:
When I was testing, I saw that when a new card was added getLastAnswer() was called twice , one with undefined and then with null while when returning back to previous card, it got called only once, with last answer. That's why I put 2 checks. Ya, maybe putting inside timeout might work, will try that.

Reply:
Done, the reason was in TutorCardDirective also, lastAnswer should have been set as the first time, that attribute was being called, fixed that now.